terrawrap/utils/config.py

```diff
@@ -86,6 +89,152 @@ def parse_wrapper_configs(wrapper_config_files: List[str]) -> WrapperConfig:
     return wrapper_config_obj
 
 
+def is_config_directory(directory: str) -> bool:
+    """
+    Checks if a wrapper file directory is a config_directory
+    :param directory: A wrapper files directory
+    :return: A boolean True if the config section of the wrapper file is not False or doesn't exist.
+    """
+    config = False
+    for file in os.listdir(directory):
+        if file.endswith(".tf"):
+            config = True
+    return config
+
+
+def has_depends_on(directory: str) -> bool:
+    """
+    Checks if a config directory has a depends_on in its tf_wrapper.
+    :param directory: The directory to check
+    :return: A boolean True if the given directory has depends_on
+    """
+    wrapper_file = os.path.join(directory, ".tf_wrapper")
+    with open(wrapper_file) as wrapper_file:
+        wrapper_config = yaml.safe_load(wrapper_file)
+        return "depends_on" in wrapper_config
+
+
+def create_wrapper_config_obj(config_dir):
+    """
+    Given a config dir containing a tf_wrapper.
+    Parses the tf_wrapper for dependencies and makes a a wrapper config object.
+    :param config_dir: A tf directory containing a tf_wrapper.
+    :return: wrapper_config_obj: a wrapper config object
+    """
+    wrapper_file = os.path.join(config_dir, ".tf_wrapper")
+    wrapper_config_obj: WrapperConfig = parse_wrapper_configs([wrapper_file])
+    if wrapper_config_obj.depends_on:
+        depends_on = []
+        for dependency in wrapper_config_obj.depends_on:
+            abs_dependency = get_absolute_path(dependency)
+            if not os.path.isdir(abs_dependency):
+                abs_dependency = get_absolute_path(dependency, config_dir)
+            depends_on.append(abs_dependency)
+        wrapper_config_obj.depends_on = depends_on
+    if not is_config_directory(os.path.dirname(wrapper_file)):
+        wrapper_config_obj.config = False
+    return wrapper_config_obj
+
+
+def walk_and_graph_directory(starting_dir: str, config_dict) -> Tuple[networkx.DiGraph, List[str]]:
+    """
+    Given a starting directory, walks it and returns all dependency info.
+    :param starting_dir: The starting directory
+    :param config_dict: A dictionary containing wrapper config objects for each seen directory
+    :return: directory_graph: A graph composed of all dependency information for a directory
+    """
+    graph_list = []
```

Comment:
Creates an empty lists of graphs, the way this works is by creating a graph of dependency information for each individual config directory and composing them all into on giant graph for an overall dependency graph